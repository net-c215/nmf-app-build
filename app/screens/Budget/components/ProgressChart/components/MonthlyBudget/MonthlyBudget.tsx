import React from "react";
import { View } from "react-native";
import styles from "./MonthlyBudget.styles";
import { Text } from "../../../../../../components";

interface Prop {
  monthlyEmissionsBudget: number;
}

const MonthlyBudget = ({ monthlyEmissionsBudget = 0 }: Prop) => (
  <View style={styles.container}>
    <Text.Secondary bold center>
      Budget for this month :{" "}
      <Text.Secondary lightGray center>
        {monthlyEmissionsBudget + " Kg"}
      </Text.Secondary>
    </Text.Secondary>
  </View>
);

export default MonthlyBudget;
