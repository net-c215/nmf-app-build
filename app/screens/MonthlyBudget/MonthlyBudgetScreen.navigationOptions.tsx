import React from "react";
import { Text } from "../../components";
import { t } from "../../utils";
import { Colors, ComponentsStyle } from "../../style";

const navigationOptions = () => ({
  headerStyle: {
    ...ComponentsStyle.header,
  },
  headerTintColor: Colors.darkLink,
  headerTitle: () => <Text.H1>{t("MONTHLY_BUDGET_TITLE")}</Text.H1>,
});

export default navigationOptions;
