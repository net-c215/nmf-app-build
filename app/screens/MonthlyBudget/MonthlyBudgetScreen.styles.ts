import { StyleSheet } from "react-native";
import colors from "../../style/colors";

const SIZE_THUMB_SLIDER = 20;

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    padding: 16,
    alignContent: "flex-start"
  },
  textContainer: {
    flexDirection: "row",
    flexWrap: "wrap"
  },
  sliderAndButtonContainer: {
    flex: 1,
    justifyContent: "space-between"
  },
  slider: {
    height: 40,
    marginTop: 14,
    marginHorizontal: 24
  },
  track: {
    height: 2,
    borderRadius: 1
  },
  thumb: {
    width: SIZE_THUMB_SLIDER,
    height: SIZE_THUMB_SLIDER,
    borderRadius: SIZE_THUMB_SLIDER / 2,
    backgroundColor: colors.linkGreen,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 2,
    shadowOpacity: 0.35
  }
});
