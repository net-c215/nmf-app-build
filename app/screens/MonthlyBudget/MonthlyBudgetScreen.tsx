import React, { useState } from "react";
import { View, Slider, ScrollView } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import * as WebBrowser from "expo-web-browser";

import { Text, Button } from "../../components";
import colors from "../../style/colors";
import styles from "./MonthlyBudgetScreen.styles";
import navigationOptions from "./MonthlyBudgetScreen.navigationOptions";
import { t } from "../../utils";

const DEFAULT_MONTHLY_CARBON_BUDGET = 500;
const MIN_MONTHLY_CARBON_BUDGET = 0;
const MAX_MONTHLY_CARBON_BUDGET = 1000;

const MonthlyBudgetScreen = ({ navigation }) => {
  const [value, setValue] = useState(DEFAULT_MONTHLY_CARBON_BUDGET);
  const onPressInfo = () =>
    WebBrowser.openBrowserAsync(
      "https://en.wikipedia.org/wiki/List_of_countries_by_carbon_dioxide_emissions_per_capita"
    );
  const onPressSaveBudget = () => navigation.goBack();

  return (
    <View style={styles.container}>
      <ScrollView>
        <View style={styles.personnalBudgetContainer}>
          <Text.Primary bold>
            {t("MONTHLY_BUDGET_MY_MONTHLY_CARBON_BUDGET")}
          </Text.Primary>
          <Text.Primary lightGray>{Math.round(value) + " kg"}</Text.Primary>
        </View>
        <Slider
          minimumTrackTintColor={colors.linkGreen}
          maximumTrackTintColor={colors.gray}
          thumbTintColor={colors.linkGreen}
          style={styles.slider}
          maximumValue={MAX_MONTHLY_CARBON_BUDGET}
          minimumValue={MIN_MONTHLY_CARBON_BUDGET}
          value={value}
          onSlidingComplete={setValue}
        />
        <View style={styles.worldBudgetContainer}>
          <View style={styles.worldExampleTitle}>
            <Text.Primary bold>
              {t("MONTHLY_BUDGET_CARBON_EMISSIONS_WORLD")}
              <Ionicons
                name="md-information-circle"
                size={26}
                color={colors.linkGreen}
                onPress={onPressInfo}
              />
            </Text.Primary>
          </View>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_LUXEMBOURG")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_UNITED_STATES")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_JAPAN")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_SWEDEN")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_FRANCE")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_CHINA")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_BRAZIL")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_INDIA")}
          </Text.Secondary>
          <Text.Secondary style={styles.worldExampleItem}>
            {t("MONTHLY_BUDGET_ETHIOPIA")}
          </Text.Secondary>
        </View>
      </ScrollView>
      <View style={styles.buttonContainer}>
        <Button.Primary onPress={onPressSaveBudget} textType={"Primary"}>
          <Text.Primary white center bold>
            {t("MONTHLY_BUDGET_SAVE")}
          </Text.Primary>
        </Button.Primary>
      </View>
    </View>
  );
};

MonthlyBudgetScreen.navigationOptions = navigationOptions;

export default MonthlyBudgetScreen;
