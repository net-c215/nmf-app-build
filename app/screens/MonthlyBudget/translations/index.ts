import en from "./en";
import fr from "./fr";
import de from "./de";

interface TranslationKeys {
  MONTHLY_BUDGET_TITLE: string;
  MONTHLY_BUDGET_MY_MONTHLY_CARBON_BUDGET: string;
}

export { en, fr, de, TranslationKeys };
