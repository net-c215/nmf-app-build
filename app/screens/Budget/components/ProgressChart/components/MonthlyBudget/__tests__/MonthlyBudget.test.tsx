import React from "react";
import renderer from "react-test-renderer";
import MonthlyBudget from "../MonthlyBudget";

const props = {
  monthlyEmissionsBudget: 1000
};

jest.unmock("../MonthlyBudget");

it("MonthlyBudget renders correctly", () => {
  const tree = renderer.create(<MonthlyBudget {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
