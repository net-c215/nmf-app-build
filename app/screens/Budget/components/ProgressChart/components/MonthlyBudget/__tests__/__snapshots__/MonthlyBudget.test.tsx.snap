// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyBudget renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 16,
      "paddingTop": 20,
    }
  }
>
  <Text.Secondary
    bold={true}
    center={true}
  >
    Budget for this month : 
    <Text.Secondary
      center={true}
      lightGray={true}
    >
      1000 Kg
    </Text.Secondary>
  </Text.Secondary>
</View>
`;
